cmd/internal/cli/build_linux.go

```diff
@@ -73,9 +78,12 @@ func fakerootExec(isDeffile bool) {
 
 	var err error
 	uid := uint32(os.Getuid())
-	if uid != 0 && !fakeroot.IsUIDMapped(uid) {
+	if buildArgs.ignoreUserns {
+		os.Setenv("APPTAINER_IGNORE_USERNS", "1")
+	} else if (uid != 0 && !fakeroot.IsUIDMapped(uid)) || buildArgs.userns {
 		sylog.Infof("User not listed in %v, trying root-mapped namespace", fakeroot.SubUIDFile)
 		os.Setenv("_APPTAINER_FAKEFAKEROOT", "1")
+		os.Setenv("APPTAINER_USERNS", "1")
 		err = fakeroot.UnshareRootMapped(args)
```

Comment:
if `buildArgs.ignoreUserns` is set then it should simulate a failure of `fakeroot.UnshareRootMapped()`.